localstack/services/cloudformation/models/iam.py

```diff
@@ -30,6 +31,14 @@ def get_physical_resource_id(self, attribute=None, **kwargs):
     def fetch_state(self, stack_name, resources):
         return IAMPolicy.get_policy_state(self, stack_name, resources, managed_policy=True)
 
+    @staticmethod
+    def add_defaults(resource, stack_name: str):
```

Comment:
nit: the code pattern in lines 36-40 (e.g., almost identical to lines 180-184, and a few other places). Wondering if we could add a small util function that we can call from the `add_defaults(..)` methods - something like:
```
def generate_default_resource_name(resource: Dict, attr_name: str, stack_name: str):
    attr_value = resource["Properties"].get(attr_name)
    if not attr_value:
        resource["Properties"][attr_value] = generate_default_name(
            stack_name, resource["LogicalResourceId"]
        )
```

Reply:
I intentionally left that out for now because I'm not sure yet if Name/LogicalId generation etc. won't be changed when the moving-refactoring of the template_deployer is done. But we can definitely add one if it survives the upcoming lifecycle refactoring :+1: